### ISSUE TITLE
Can we make documentation apply the "Keep It Stupid Simple" principle?

### ISSUE THREAD

**Issue by kevingates** [IdentityServer/IdentityServer4#3178]
### Question / Steps to reproduce the problem
Code examples on documentation page are ambiguous for beginners.

For example, compare this page: 
1. https://identityserver4.readthedocs.io/en/latest/quickstarts/1_client_credentials.html
to this page:
2. https://identityserver4.readthedocs.io/en/latest/quickstarts/2_resource_owner_passwords.html

You will notice the second one (passwords) has example code, but doesn't say where to put the sample code. In short, it presumes a lot. Other pages do this as well.

### Suggestions for Improvement
As one who had worked as a manager of customer support and led documentation efforts, I strongly encourage (and honestly , would greatly benefit from) an effort to update all coding examples to explicitly state **where** the code is to be applied each time. (PS: I know this can't happen everywhere immediately.)

Sometimes this could be added as follows:
______
> Add the code below to  **src > api > Controllers > IdentityController.cs**
> ```
> // Some code
>```
>On your next step, you will need to edit the **Startup** file. Go to : **src > app > Startup.cs**. Then go under the **ConfigureServices** method and add:
>` // Your other code `

In short, for people like me, new to backend work, especially in dotnet, it would be helpful if all documentation includes every step and presumes nothing.

The benefit of this is two-fold:
1. It allows newcomers to learn quickly, and therefore also able to become users of IdentityServer, and possibly contributors one day.
2. It helps the current contributors to consistently work through every step, and possible help proactively "debug" and provide support. If they can't think of a simple way to convey something, then likely they need assistance -- which is fine. A solution will be made eventually that is better for everyone.

**Comment by redplane** [comment]
Same issue.
The tutorial is very hard to beginners. Especially `authorization_code`, for now, I'm struggling with some issues that have been logged to this repo.

**Comment by leastprivilege** [comment]
I would very much appreciate if someone would spend the time to make the docs better. But we are both very busy with "real" work right now.

If you - or @LindaLawton - want to take a stab at it, that would be cool.

The "reference" code can be found here:
https://github.com/IdentityServer/IdentityServer4.Samples/tree/master/Quickstarts

**Comment by LindaLawton** [comment]
I am happy to rewrite some of the tutorials.

**Comment by LindaLawton** [comment]
> note:: The protocol used in this Template is ``http`` and the port is set to 5000 when running on Kestrel or a random one on IISExpress. You can change that in the ``Properties\launchSettings.json`` file. However, all of the quickstart instructions will assume you use the default port on Kestrel as well as the ``http`` protocol, which is sufficient for local development.

Do we still want to use HTTP instead of setting this up to run HTTPS from the start?

**Comment by kevingates** [comment]
@LindaLawton I appreciate the work you're doing to help people like me to start understanding and hopefully using Identity Server.

Also, I vote if possible for all guides to use HTTPS since that is likely the setup for every user of IS unless there are significant / material reasons to stick with HTTP only.

**Comment by kevingates** [comment]
Any updates on this?

**Comment by kevingates** [comment]
Any updates on this?
